src/ndt-client.js

```diff
@@ -62,10 +62,86 @@ var NDTjs = function(serverAddress, serverPort, serverPath, verboseDebug) {
 
   this.logger('Initialized NDTjs with the following settings: ' +
       JSON.stringify(this.settings));
+
+  if (!this.checkBrowserSupport()) {
+    this.logger('Browser or runtime environment does not support WebSockets');
+    throw new Error('UnsupportedBrowser');
+  }
 };
 
+/**
+ * Log a message to the console if debugging is enabled.
+ *
+ * @param {String} logMessage - The message to log.
+ */
+
 NDTjs.prototype.logger = function(logMessage) {
+
   if (this.settings.verboseDebug) {
     console.log(logMessage);
   }
 };
+
+/**
+ * Check that the browser supports the NDT test.
+ *
+ * @returns {Boolean} Browser supports necessary functions for test client.
+ */
+
+NDTjs.prototype.checkBrowserSupport = function() {
+
+  if (WebSocket === undefined && window.WebSocket === undefined &&
+      window.MozWebSocket === undefined) {
+    return false;
+  }
+  return true;
+};
+
+/**
+ * Parses messages received from the NDT server.
+ *
+ * @param {Buffer} passedBuffer Raw message received from the NDT server.
+ * @returns {Object} Parsed messaged with type and message properties.
+ */
+
+NDTjs.prototype.parseNDTMessage = function(passedBuffer) {
```

Comment:
Wrong.

The first byte is the type, the next two bytes are the high-order and low-order bits of the size of the message.

For your test message, [1, 0, 2, 'O', 'K'] would be a good test input.